File: nprimes.c
/* 
 * get number of primes smaller than 1000000
 *
 * compile with -lm option to user math library
 *
 * */

#include <stdio.h>
#include <math.h>

int isprime(int n){
    int i;
    if (n<=1) return 0;
    if (n==2) return 1;
    if (n%2==0) return 0;
    for (i=sqrt(n)+1;i>1;i--){
        if (n%i==0)
            return 0;
    }
    return 1;
}

int main(int argc, char** argv){
    int mx = 10000000;     /* mx=8000 => cnt=1007 */
    int i;
    int cnt = 0;

    for (i=0;i<mx;i++){
        if (isprime(i))
            cnt++;
    }

    printf("Number of primes smaller than %d:%d",mx,cnt);
    return 0;
}

File: nprimes.lua
function isprime(n)
    if n==1 then
        return false
    end
    if n==2 then
        return true
    end
    if n%2==0 then
        return false
    end
    c = math.floor((math.sqrt(n))+1)
    while n%c  ~= 0 do
        c = c-1
        if c==1 then
            return true
        end
    end
    return false
end

cnt = 0
for i=1,10000 do
    if isprime(i) then
        cnt = cnt + 1
    end
end
print (cnt)
